doc/classes/vector.adoc

```diff
@@ -0,0 +1,256 @@
+Returns a `number` which is equal to the angle (in radians) between `a` and `b`.
+
+Formula used is stem:[tan^-1(|bar a xx bar b|, bar a * bar b)].
+
+=== `vector.dot(a, b) -> number`
+* `a`, `b`: `vector`
+
+Returns a `number` equal to the dot product of `a` and `b`.
+
+=== `vector.cross(a, b) -> vector`
+* `a`, `b`: `vector`
+
+Returns a new `vector` which is equal to the cross product of `a` and `b`.
+
+=== `vector.add(a, b) -> vector`
+* `a`: `vector`
+* `b`: `vector` or `number`
+
+If `b` is a `vector`:: Returns a new `vector` where each component of `a` is added to each component of `b`
+If `b` is a `number`:: Returns a new `vector` where each component of `a` is added to `b`
+
+=== `vector.subtract(a, b) -> vector`
+* `a`: `vector`
+* `b`: `vector` or `number`
+
+If `b` is a `vector`:: Returns a new `vector` where each component of `a` is subtracted from each component of `b`
+If `b` is a `number`:: Returns a new `vector` where each component of `a` is subtracted from `b`
+
+=== `vector.multiply(a, b) -> vector`
+* `a`: `vector`
+* `b`: `vector` or `number`
+
+If `b` is a `vector`:: Returns a new `vector` where each component of `a` is multiplied by component of `b`
+If `b` is a `number`:: Returns a new `vector` where each component of `a` is multiplied by `b`
+
+=== `vector.divide(a, b) -> vector`
+* `a`: `vector`
+* `b`: `vector` or `number`
+
+If `b` is a `vector`:: Returns a new `vector` where each component of `a` is divided by component of `b`
+If `b` is a `number`:: Returns a new `vector` where each component of `a` is divided by `b`
+
+=== `vector.offset(v, x, y, z) -> vector`
+* `v`: `vector`
+* `x`, `y`, `z`: `number`
+
+Returns a new `vector` where each component of `x`, `y`, and `z` are added to the respective components of `v`.
+
+Equivalent to `vector.add(v, {x = x, y = y, z = z})`.
+
+=== `vector.sort(a, b) -> vector, vector`
+* `a`: `vector`
+* `b`: `vector`
+
+Returns two new `vector` values.
+
+The first consists of the smaller components of `a` and `b`, where each is equal to `math.min(a.N, b.N)` (N being one of x, y, or z).
+
+The second is similar to the first, but consisting of the larger components instead.
+
+=== `vector.check(v) -> bool`
+* `v`: `vector`
+
+Returns `true` if `v` is a valid, metatable-enhanced vector.
+
+Returns `false` otherwise.
+
+=== `vector.rotate_around_axis(v, axis, angle) -> vector`
+* `v`, `axis`: `vector`
+* `angle`: `number`
+
+Returns a new `vector` which is equal to `v` rotated around `axis` by `angle` radians.
+
+=== `vector.rotate(v, rot) -> vector`
+* `v`, `rot`: `vector`
+
+Returns a new `vector` which is equal to `v` rotated by `rot`.
+
+The way that the components of `rot` map is as follows:
+
+* `rot.x` is pitch
+* `rot.y` is yaw
+* `rot.z` is roll
+
+=== `vector.dir_to_rotation(forward, up) -> vector`
+* `forward`: `vector`
+* `up`: `vector` or `nil`
+
+Returns a new rotational `vector` (the same kind as `rot` in `vector.rotate()`) equal to the rotation from `up` to `forward`.
```

Comment:
Not a fan of the `func()` syntax used in the Lua API. Either (1) omit the brackets or (2) add valid arguments. Currently this is a call without args to me.